Playmate/Models/PlayerConnection.m

```diff
@@ -11,15 +11,107 @@ @implementation PlayerConnection
 
 @dynamic userObjectId;
 @dynamic connections;
+@dynamic friendsList;
+@dynamic pendingList;
 
 + (nonnull NSString *)parseClassName {
     return @"PlayerConnection";
 }
 
-+ (void)savePlayer:(NSString *)objectId withConnections:(NSDictionary *)connect {
+//+ (void)savePlayer:(NSString *)objectId withConnections:(NSDictionary *)connect {
+//    PlayerConnection *pc = [PlayerConnection new];
+//    pc.userObjectId = objectId;
+//    pc.connections = connect;
+//
+//    [pc saveInBackground];
+//}
+
++ (PlayerConnection *)initializePlayerConnection {
+    
     PlayerConnection *pc = [PlayerConnection new];
-    pc.userObjectId = objectId;
-    pc.connections = connect;
+    
+    PFUser *me = [PFUser currentUser];
+    pc.userObjectId = me.objectId;
+    pc.connections = [NSMutableDictionary new];
+    pc.friendsList = [NSMutableArray new];
+    pc.pendingList = [NSMutableArray new];
+    
+    [pc saveInBackground];
+    return pc;
+}
+
+// for saving own(A) connection to someone(B) else; Save B in A's dictionary (A is me)
++ (void)saveMyConnectionTo:(NSString *)otherObjectId withStatus:(BOOL)areFriends andWeight:(int)weight {
+    
+//    ConnectionState *cs = [ConnectionState new];
+//    cs.areFriends = areFriends;
+//    cs.relationshipWeight = weight;
+    
+    PFUser *me = [PFUser currentUser];
+    [me fetchIfNeeded];
+    
+    PlayerConnection *pc = me[@"playerConnection"][0];
+    [pc fetchIfNeeded];
+
+//    [pc.connections setObject:cs forKey:otherObjectId];
+    NSLog(@"pc = %@", pc[@"friendsList"]);
```

Comment:
Let's avoid over using NSLog, but let's also discuss a way to give them a more understanding format